### PR TITLE
[AD-974] Allow the option allowDiskUse to be set via an environment variable.

### PR DESCRIPTION
**Pull request by birschick-bq**
### Summary

[AD-974] Allow the option allowDiskUse to be set via an environment variable.

### Description

- Adds the allowDiskUse connection option.
- Adds ability to set options via the DOCUMENTDB_CUSTOM_OPTIONS environment variable.

### Related Issue

https://bitquill.atlassian.net/browse/AD-974

### Additional Reviewers
@affonsoBQ
@alinaliBQ
@andiem-bq
@alexey-temnikov
@birschick-bq
@mitchell-elholm

### CODE REVIEW

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbQueryExecutorTest.java [hunk 1]
```diff
@@ -77,18 +78,17 @@ void initialize() throws SQLException {
...
-                                null,
-                                0,
-                                0));
+        executor = new MockQueryExecutor(
+                statement,
+                VALID_CONNECTION_PROPERTIES,
+                null,
+                0,
+                0);
+        statement = new DocumentDbStatement(connection, executor);
```
Comment (alinaliBQ) [inline src/test/java/software/amazon/documentdb/jdbc/DocumentDbQueryExecutorTest.java:91]: should it be concerned that executor and statement depend on each other (i.e., executor is created using statement as input and vise versa)?

File: src/main/java/software/amazon/documentdb/jdbc/DocumentDbConnectionProperties.java [hunk 2]
```diff
@@ -912,6 +936,22 @@ public static DocumentDbConnectionProperties getPropertiesFromConnectionString(
...
+        }
+        final String[] propertyPairs = customOptions.split(";");
+        for (String pair : propertyPairs) {
+            final int splitIndex = pair.indexOf("=");
+            final String key = pair.substring(0, splitIndex);
+            final String value = pair.substring(1 + splitIndex);
+
+            addPropertyIfValid(properties, key, value, true, true);
+        }
+
```
Comment (alinaliBQ) [inline src/main/java/software/amazon/documentdb/jdbc/DocumentDbConnectionProperties.java]: nit empty line

File: src/markdown/setup/connection-string.md [hunk 3]
```diff
@@ -16,28 +16,29 @@
...
+| `allowDiskUse`             | (enum/string) Provides the option for using the `allowDiskUse` query option globally on all queries. A value of `default` indicates that the system setting should be used. A value of `enable` or `disable` sets the option to `true` or `false`, respectively, on the query option for all queries.                                                                                                                                                                                                                                                                                                                                                                                                                                       | `enable`
```
Comment (affonsoBQ) [inline src/markdown/setup/connection-string.md:41]: missing the alternative to set this.
Will we document how the custom_options env works?

**Review by affonsov — APPROVED** [review]

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbQueryExecutorTest.java (hunk 1 above)
Comment (birschick-bq) [reply src/test/java/software/amazon/documentdb/jdbc/DocumentDbQueryExecutorTest.java:91]: Indeed this is a circular reference. Luckily, the GC handles this when cleaning up. This overload is just for testing. Normally, the `DocumentDbQueryExecutor` is created inside the `DocumentDbStatement` constructor.

File: src/main/java/software/amazon/documentdb/jdbc/DocumentDbConnectionProperties.java (hunk 2 above)
Comment (birschick-bq) [reply src/main/java/software/amazon/documentdb/jdbc/DocumentDbConnectionProperties.java]: Fixed.

**Review by alinaliBQ — COMMENTED** [review]
Looks good to me!

**Pull request merged**